ark/utils/deepcell_service_utils_test.py

```diff
@@ -22,87 +28,103 @@ def mocked_run_deepcell(input_dir, output_dir, host, job_type, scale):
 
-        # make sure DeepCell (.zip) output exists
-        assert os.path.exists(os.path.join(output_dir, 'example_output.zip'))
+        with tempfile.TemporaryDirectory() as output_dir:
 
-        # DeepCell output .zip file should be extracted
-        assert os.path.exists(os.path.join(output_dir, 'fov1_feature_0.tif'))
-        assert os.path.exists(os.path.join(output_dir, 'fov2_feature_0.tif'))
-        assert os.path.exists(os.path.join(output_dir, 'fov3_feature_0.tif'))
+            # if fovs is None, all .tif files in input dir should be taken
+            create_deepcell_output(deepcell_input_dir=input_dir, deepcell_output_dir=output_dir)
 
-        os.remove(os.path.join(output_dir, 'fov1_feature_0.tif'))
-        os.remove(os.path.join(output_dir, 'fov2_feature_0.tif'))
-        os.remove(os.path.join(output_dir, 'fov3_feature_0.tif'))
-        os.remove(os.path.join(output_dir, 'example_output.zip'))
+            # make sure DeepCell (.zip) output exists
+            assert os.path.exists(os.path.join(output_dir, 'example_output.zip'))
 
-        # if fovs is None, all .tif files in input dir should be taken
-        create_deepcell_output(deepcell_input_dir=input_dir, deepcell_output_dir=output_dir)
+            assert os.path.exists(os.path.join(output_dir, 'fov1_feature_0.tif'))
+            assert os.path.exists(os.path.join(output_dir, 'fov2_feature_0.tif'))
+            assert os.path.exists(os.path.join(output_dir, 'fov3_feature_0.tif'))
 
-        # make sure DeepCell (.zip) output exists
-        assert os.path.exists(os.path.join(output_dir, 'example_output.zip'))
+            pathlib.Path(os.path.join(input_dir, 'fovs.zip')).touch()
 
-        assert os.path.exists(os.path.join(output_dir, 'fov1_feature_0.tif'))
-        assert os.path.exists(os.path.join(output_dir, 'fov2_feature_0.tif'))
-        assert os.path.exists(os.path.join(output_dir, 'fov3_feature_0.tif'))
+            # Warning should be displayed if fovs.zip file exists (will be overwritten)
+            with pytest.warns(UserWarning):
+                create_deepcell_output(deepcell_input_dir=input_dir,
+                                       deepcell_output_dir=output_dir, fovs=['fov1'])
 
-        pathlib.Path(os.path.join(input_dir, 'fovs.zip')).touch()
+            # DeepCell output .tif file does not exist for some fov
+            with pytest.warns(UserWarning):
+                create_deepcell_output(deepcell_input_dir=input_dir,
+                                       deepcell_output_dir=output_dir, suffix='_other_suffix',
+                                       fovs=['fov1'])
 
-        # Warning should be displayed if fovs.zip file exists (will be overwritten)
-        with pytest.warns(UserWarning):
-            create_deepcell_output(deepcell_input_dir=input_dir, deepcell_output_dir=output_dir,
-                                   fovs=['fov1'])
+            pathlib.Path(os.path.join(input_dir, 'fov4.tif')).touch()
```

Comment:
Is there a specific warning that we're making sure is raised here, or is this just so that we now have 4 TIFs for the batch size checks that come next?

Reply:
the last `pytest.warns(UserWarning)` is redundantly checking warnings for 'overwriting file'; definitely could be removed if needed.  The extra fov is created for the batch size checking later.